drivers/clock_control/clock_control_mcux_syscon.c

```diff
@@ -33,6 +33,28 @@ static int mcux_lpc_syscon_clock_control_on(const struct device *dev,
 	}
 #endif /* defined(CONFIG_COUNTER_NXP_MRT) */
 
+#if defined(CONFIG_PINCTRL_NXP_KINETIS)
+	switch ((uint32_t)sub_system) {
+	case MCUX_PORT0_CLK:
+		CLOCK_EnableClock(kCLOCK_Port0);
+		break;
+	case MCUX_PORT1_CLK:
+		CLOCK_EnableClock(kCLOCK_Port1);
+		break;
+	case MCUX_PORT2_CLK:
+		CLOCK_EnableClock(kCLOCK_Port2);
+		break;
+	case MCUX_PORT3_CLK:
+		CLOCK_EnableClock(kCLOCK_Port3);
+		break;
+	case MCUX_PORT4_CLK:
+		CLOCK_EnableClock(kCLOCK_Port4);
+		break;
+	default:
+		break;
+	}
+#endif /* defined(CONFIG_PINCTRL_NXP_KINETIS) */
```

Comment:
if we are putting here the ports and the flexcomms, can't we also put the GPIOs?

Reply:
The GPIO driver needs to be updated for this which is beyond the scope of this PR and can be addressed as a separate PR.

Reply:
the gpio driver is already being updated significantly in this PR?

Reply:
However adding clocking would involve updating other SoC's which use this driver and I do not want to address it in this PR.

Reply:
ok, fine, not a  big deal